### ISSUE TITLE
Distributed Background Tasks

### ISSUE THREAD

**Issue by MikeAlhayek** [OrchardCMS/OrchardCore#13219]
### Is your feature request related to a problem? Please describe.
Currently, Background Tasks run on local machine. When scaling the app to multiple instances, running a background task per instance isn't always a good approach. For example, sending notification, billing.

### Describe the solution you'd like
Add a way to prevent Background tasks from running on multiple instances. I believe @jtkech started something called jobs that does this.

Anyhow, maybe we can add a new property called `IsJob`  to `BackgroundTaskAttribute` that would allow us to run background tasks in a distributed fashion instead of per instance.

**Comment by jtkech** [comment]
Yes, we have the tools but still need custom implementations for example to manage the ownership of a given task. Not directly related but currently I'm tweaking `DistributedShellHostedService`, so while it's hot I hope I will be able to start on this soon, then we could see what would be good to have on the UI side.

Related #5755

**Comment by sebastienros** [comment]
> maybe we can add a new property called IsJob

No we can't. It's a Job, not a background task. Developers will create background tasks or they will create jobs. We don't want to change the behavior of a "task" that is design to work for one case or the other.

**Comment by MikeAlhayek** [comment]
@jtkech any update on the jobs? @sebastienros mentioned this was designed while back and it should have been implemented. Any idea? Is it still incomplete work? Anything I can help with?

**Comment by jtkech** [comment]
No problem, I'll try to work on it soon and will let you know

**Comment by MikeAlhayek** [comment]
Thank you! Let me know if I can help
